src/tools/schema2packager.ts

```diff
@@ -163,12 +175,25 @@ ${name} clone_entity(const ${name}& entity) {
 }
 
 template<>
-void decode_entity(${name}* entity, const char* str) {
+bool entities_equal(const ${name}& a, const ${name}& b) {
+  return ${equals.join('\n         ')} 1;
```

Comment:
What's this 1 doing here??

Reply:
OK I figured it out.. That's pretty subtle.

How about `equals.join('\n      &&')` ?

Reply:
Updated to put the `&&` s in that line.